File: Easy/Two_Sum.cpp
#include <vector>
#include <iostream>
#include <math.h>
 
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> answer;
        bool t = false;
       
        
        if(t == false){
            for(int i = 0; i < nums.size(); i++){
            for(int j = i + 1; j < nums.size(); j++){
                    if(nums.at(i) + nums.at(j) == target){
                        answer.push_back(i);
                        answer.push_back(j);
                        t = true;
                        break;
                    }
                }
            
            if(t){
                break;
            }
            }
        }
    return answer;
    }
};

File: Easy/Two_Sum.java
import java.util.HashMap;
import java.util.Map;

class Solution {    // put nums into a map and save index and then check for remained after subtracting the target
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> myMap = new HashMap<>();
        int n = nums.length;

        
        for (int i = 0; i < n; i++) {
            myMap.put(nums[i], i);
        }

       
        for (int i = 0; i < n; i++) {
            int remainder = target - nums[i];
            if (myMap.containsKey(remainder) && myMap.get(remainder) != i) {
                return new int[]{i, myMap.get(remainder)};
            }
        }

        return new int[]{}; // No solution found
    }
}